src/carousel.tsx

```diff
@@ -0,0 +1,727 @@
+import * as React from 'react';
+import IconChevronLeftRegular from './generated/mistica-icons/icon-chevron-left-regular';
+import IconChevronRightRegular from './generated/mistica-icons/icon-chevron-right-regular';
+import {useScreenSize, useTheme} from './hooks';
+import Inline from './inline';
+import {createUseStyles} from './jss';
+import Stack from './stack';
+import Touchable from './touchable';
+import classNames from 'classnames';
+import {useResonsiveLayoutMargin} from './responsive-layout';
+import {useIsInverseVariant, ThemeVariant} from './theme-variant-context';
+import {applyAlpha} from './utils/color';
+import {DisableBorderRadiusProvider} from './image';
+import {getPrefixedDataAttributes} from './utils/dom';
+
+import type {DataAttributes} from './utils/types';
+import type {Theme} from './theme';
+
+const listenResize = (element: Element, handler: () => void) => {
+    const getResizeObserverPromise = (): Promise<typeof window.ResizeObserver> => {
+        return window.ResizeObserver
+            ? Promise.resolve(ResizeObserver)
+            : import(/* webpackChunkName: "@juggle/resize-observer" */ '@juggle/resize-observer').then(
+                  (m: any) => m.ResizeObserver
+              );
+    };
+    let cancelled = false;
+    let resizeObserver: ResizeObserver | null = null;
+    getResizeObserverPromise().then((ResizeObserver) => {
+        if (!cancelled) {
+            resizeObserver = new ResizeObserver(handler);
+            resizeObserver.observe(element);
+        }
+    });
+
+    return () => {
+        cancelled = true;
+        if (resizeObserver) {
+            resizeObserver.disconnect();
+        }
+    };
+};
```

Comment:
maybe this could be used https://github.com/Telefonica/mistica-web/blob/afc94c39343c4e4ecf9098a63ea899731afe43fe/src/hooks.tsx#L65?

or create a common hook or something?

Reply:
I could create a common util function, yes